scipy/stats/stats.py

```diff
@@ -4226,6 +4232,395 @@ def weightedtau(x, y, rank=True, weigher=None, additive=True):
+
+    Notes
+    -----
+
+    A description of the process of MGC and applications on neuroscience data
+    can be found in [1]_. It is performed using the following steps:
+
+    #. Two distance matrices :math:`A` and :math:`B` are computed and
+       modified to be mean zero columnwise. This results in two
+       :math:`n \times n` distance matrices :math:`C^x` and :math:`C^y` (the
+       centering and unbiased modification) [3]_.
+
+    #. For all values :math:`k` and :math:`l` from :math:`1, ..., n`,
+
+       #. The :math:`k`-nearest neighbor and :math:`l`-nearest neighbor graphs
+          are calculated for each property. Here, :math:`G_k (i, j)` indicates
+          the :math:`k`-smallest values of the :math:`i`-th row of :math:`A`
+          and :math:`H_l (i, j)` indicates the :math:`l` smallested values of
+          the :math:`i`-th row of :math:`B`
+
+       #. Let :math:`\circ` denotes the entry-wise matrix product, then local
+          correlations are summed and normalized using the following statistic:
+
+    .. math::
+
+        c^{kl} = \frac{\sum_{ij} A G_k B H_l}
+                      {\sqrt{\sum_{ij} A^2 G_k \times \sum_{ij} B^2 H_l}}
+
+    #. The MGC test statistic is the smoothed optimal local correlation of
+       :math:`\{ c^{kl} \}`. Denote the smoothing operation as :math:`R(\cdot)`
+       (which essentially set all isolated large correlations) as 0 and
+       connected large correlations the same as before, see [3]_.) MGC is,
+
+    .. math::
+
+        MGC_n (x, y) = \max_{(k, l)} R \left(c^{kl} \left( x_n, y_n \right)
+                                                    \right)
+
+    The test statistic returns a value between :math:`(-1, 1)` since it is
+    normalized.
+
+    The p-value returned is calculated using a permutation test. This process
+    is completed by first randomly permuting :math:`y` to estimate the null
+    distribution and then calculating the probability of observing a test
+    statistic, under the null, at least as extreme as the observed test
+    statistic.
+
+    MGC requires at least 5 samples to run with reliable results. It can also
+    handle multidimensional arrays.
+
+    .. versionadded:: 1.4.0
+
+    References
+    ----------
+    .. [1] Vogelstein, J. T., Bridgeford, E. W., Wang, Q., Priebe, C. E.,
+           Maggioni, M., & Shen, C. (2019). Discovering and deciphering
+           relationships across disparate data modalities. ELife.
+    .. [2] Panda, S., Palaniappan, S., Xiong, J., Swaminathan, A.,
+           Ramachandran, S., Bridgeford, E. W., ... Vogelstein, J. T. (2019).
+           mgcpy: A Comprehensive High Dimensional Independence Testing Python
+           Package. ArXiv:1907.02088 [Cs, Stat].
+    .. [3] Shen, C., Priebe, C.E., & Vogelstein, J.T. (2019). From distance
+           correlation to multiscale graph correlation. Journal of the American
+           Statistical Association.
+
+    Examples
+    --------
+    >>> import numpy as np
```

Comment:
I think we can safely assume this import has already happened for our refguide_check infrastructure

Reply:
indeed, this line should be left out of all examples

Reply:
https://github.com/scipy/scipy/pull/10524/commits/a97997aa164c6b6f2d737e41e4e3a4f59a9ad9ba